redash/handlers/queries.py

```diff
@@ -264,3 +265,23 @@ def post(self, query_id):
         parameter_values = collect_parameters_from_request(request.args)
 
         return run_query(query.data_source, parameter_values, query.query_text, query.id)
+
+
+class QueryVersionListResource(BaseResource):
+    @require_permission('view_query')
+    def get(self, query_id):
+        results = models.Change.query.filter(
+            models.Change.object_type == 'queries',
+            models.Change.object_id == query_id)
+        page = request.args.get('page', 1, type=int)
+        page_size = request.args.get('page_size', 25, type=int)
+        return paginate(results, page, page_size, lambda c: c.to_dict(full=False))
+
+
+class QueryVersionResource(BaseResource):
+    @require_permission('view_query')
+    def get(self, query_id, version_id):
```

Comment:
Why not use the `Change` object's primary key? It will make the query much more efficient and simpler.